src/transcode.jl

```diff
@@ -121,6 +84,95 @@ function Base.transcode(codec::Codec, data::ByteData)
     throw(error[])
 end
 
+"""
+    transcode(codec::Codec, data::ByteData)::ByteData
+
+Transcode `data` by applying `codec`.
+
+Note that this method does not initialize or finalize `codec`. This is
+efficient when you transcode a number of pieces of data, but you need to call
+[`TranscodingStreams.initialize`](@ref) and
+[`TranscodingStreams.finalize`](@ref) explicitly.
+
+Examples
+--------
+
+```julia
+julia> using CodecZlib
+
+julia> data = b"abracadabra";
+
+julia> codec = ZlibCompressor();
+
+julia> TranscodingStreams.initialize(codec)
+
+julia> compressed = transcode(codec, data);
+
+julia> TranscodingStreams.finalize(codec)
+
+julia> codec = ZlibDecompressor();
+
+julia> TranscodingStreams.initialize(codec)
+
+julia> decompressed = transcode(codec, compressed);
+
+julia> TranscodingStreams.finalize(codec)
+
+julia> String(decompressed)
+"abracadabra"
+
+```
+"""
+Base.transcode(codec::Codec, data::ByteData) = transcode(codec, Buffer(data))
+
+Base.transcode(codec::Codec, data::ByteData, output::Buffer) =
+    transcode(codec, Buffer(data), output)
+
+"""
+    transcode(codec::Codec, data::ByteData, output::ByteData)::ByteData
```

Comment:
Could we consolidate the docstring into a single docstring with multiple signatures?